integrations/googledrive/src/files-cache.ts

```diff
@@ -0,0 +1,86 @@
+import { RuntimeError, z } from '@botpress/sdk'
+import { baseGenericFileSchema } from './schemas'
+import { BaseFolderFile, BaseGenericFile, BaseNormalFile } from './types'
+import * as bp from '.botpress'
+
+const filesMapImpl = z.record(z.string(), baseGenericFileSchema)
+type FilesMap = z.infer<typeof filesMapImpl>
+
+export class FilesCache {
+  private _map: FilesMap
+  public constructor(private _client: bp.Client, private _ctx: bp.Context) {
+    this._map = {}
+  }
+
+  public static async load({ client, ctx }: { client: bp.Client; ctx: bp.Context }): Promise<FilesCache> {
+    const getStateResponse = await client.getOrSetState({
+      id: ctx.integrationId,
+      type: 'integration',
+      name: 'list',
+      payload: {
+        filesMap: JSON.stringify({}),
+      },
+    })
+    const parseResult = filesMapImpl.safeParse(JSON.parse(getStateResponse.state.payload.filesMap))
+    if (parseResult.error) {
+      throw new RuntimeError(`Error parsing saved files map: ${parseResult.error.message}`)
+    }
+    const cachedMap = parseResult.data
+    const map = new FilesCache(client, ctx)
+    map._map = cachedMap
+    return map
+  }
+
+  public async save() {
+    await this._client.setState({
+      id: this._ctx.integrationId,
+      type: 'integration',
+      name: 'list',
+      payload: {
+        filesMap: JSON.stringify(this._map),
+      },
+    })
+  }
+
+  public has(id: string): boolean {
```

Comment:
**nitpick:** instead of `has()` + `get()` I often prefer a `find()` method that returns `T | undefined`